tests/subsys/openthread/radio_test.c

```diff
@@ -0,0 +1,818 @@
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+	ztest_check_expected_value(duration);
+	scan_done_cb = done_cb;
+	return ztest_get_return_value();
+}
+
+static int rssi_scan_mock(struct device *dev, u16_t duration,
+			  energy_scan_done_cb_t done_cb)
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+	zassert_equal(duration, 1,
+		      "otPlatRadioGetRssi shall pass minimal allowed value.");
+
+	/* use return value as callback param */
+	done_cb(&radio, ztest_get_return_value());
+
+	return 0;
+}
+
+static int set_channel_mock(struct device *dev, u16_t channel)
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+	ztest_check_expected_value(channel);
+	return ztest_get_return_value();
+}
+
+void otPlatRadioEnergyScanDone(otInstance *aInstance, int8_t aEnergyScanMaxRssi)
+{
+	zassert_equal(aInstance, ot, "Incorrect instance.");
+	ztest_check_expected_value(aEnergyScanMaxRssi);
+}
+
+void otSysEventSignalPending(void)
+{
+	k_sem_give(&ot_sem);
+}
+
+void otTaskletsSignalPending(otInstance *aInstance)
+{
+	zassert_equal(aInstance, ot, "Incorrect instance.");
+	k_sem_give(&ot_sem);
+}
+
+static void make_sure_sem_set(k_timeout_t timeout)
+{
+	zassert_equal(k_sem_take(&ot_sem, timeout), 0, "Sem not released.");
+}
+
+void otPlatRadioReceiveDone(otInstance *aInstance, otRadioFrame *aFrame,
+			    otError aError)
+{
+	zassert_equal(aInstance, ot, "Incorrect instance.");
+	ztest_check_expected_value(aFrame->mChannel);
+	ztest_check_expected_value(aFrame->mLength);
+	ztest_check_expected_data(aFrame->mPsdu, aFrame->mLength);
+	ztest_check_expected_value(aError);
+}
+
+void otPlatRadioTxDone(otInstance *aInstance, otRadioFrame *aFrame,
+		       otRadioFrame *aAckFrame, otError aError)
+{
+	zassert_equal(aInstance, ot, "Incorrect instance.");
+	ztest_check_expected_value(aError);
+}
+
+static enum ieee802154_hw_caps get_capabilities(struct device *dev)
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+
+	return IEEE802154_HW_FCS | IEEE802154_HW_2_4_GHZ |
+	       IEEE802154_HW_TX_RX_ACK | IEEE802154_HW_FILTER |
+	       IEEE802154_HW_ENERGY_SCAN;
+}
+
+static enum ieee802154_hw_caps get_capabilities_caps_mock(struct device *dev)
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+
+	return ztest_get_return_value();
+}
+
+static int configure_mock(struct device *dev, enum ieee802154_config_type type,
+			  const struct ieee802154_config *config)
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+
+	zassert_equal(type, IEEE802154_CONFIG_EVENT_HANDLER,
+		      "Only event handler configuration was expected.");
+
+	return 0;
+}
+
+static int configure_match_mock(struct device *dev,
+				enum ieee802154_config_type type,
+				const struct ieee802154_config *config)
+{
+	zassert_equal(dev, &radio, "Device handle incorrect.");
+	ztest_check_expected_value(type);
+
+	switch (type) {
+	case IEEE802154_CONFIG_AUTO_ACK_FPB:
+		ztest_check_expected_value(config->auto_ack_fpb.enabled);
```

Comment:
`auto_ack_fpb` was extended with `mode` field, you chould verify this as well (should be `IEEE802154_FPB_ADDR_MATCH_THREAD`).